ale_linters/elm/make.vim

```diff
@@ -0,0 +1,64 @@
+" Author: buffalocoder - https://github.com/buffalocoder
+
+function! ale_linters#elm#make#Handle(buffer, lines)
+    let l:output = []
+    for l:line in a:lines
+        if l:line[0] ==# '['
+            let l:errors = json_decode(l:line)
+
+            for l:error in l:errors
+                call add(l:output, {
+                \    'bufnr': a:buffer,
+                \    'lnum': l:error.region.start.line,
+                \    'vcol': 0,
+                \    'col': l:error.region.start.column,
+                \    'type': (l:error.type ==? 'error') ? 'E' : 'W',
+                \    'text': l:error.overview,
+                \    'nr': -1,
+                \})
+            endfor
+        endif
+    endfor
+
+    return l:output
+endfunction
+
+" This function was copied from from https://github.com/ElmCast/elm-vim.
+" All credit goes to that project!
+" Returns the closest parent with an elm-package.json file.
+function! s:FindRootDirectory() abort
+    let l:elm_root = getbufvar('%', 'elmRoot')
+    if empty(l:elm_root)
+        let l:current_file = expand('%:p')
+        let l:dir_current_file = fnameescape(fnamemodify(l:current_file, ':h'))
+        let l:match = findfile('elm-package.json', l:dir_current_file . ';')
```

Comment:
This search can fail when the callback fires from a different buffer from the buffer the linter is being run for. You should try using the `FindFile` function in the util module. That will probably do the job.